### PR TITLE
Set cookie for setup and snapshot

### PR DESCRIPTION
**Pull request by tasleson**
Previous PR:
#323

Introduced a cookie to get userspace to handle udev event, but
we left out the setup and snapshot code paths, so add those too.

### CODE REVIEW

**Comment by agrover** [comment]
The second commit looks fine, everybody likes more tests. For the first commit, well, it includes two things: setting the flag for snapshot/setup/new but also some refactoring. Clearly we need to do the first thing but the refactoring seems like a marginal win -- lines of code and and code paths go up.

Do you mind splitting the first commit into two, one that fixes the bug, and then one that does the refactoring?

**Comment by tasleson** [comment]
The re-factor was done to reduce the number of code paths that are almost a direct copy & paste.  I very much subscribe to: https://en.wikipedia.org/wiki/Don%27t_repeat_yourself , but I will re-work to just fix the issue as you request.

**Comment by tasleson** [comment]
I'll put the re-factoring into a different PR.

**Comment by agrover** [comment]
Thanks. Yeah, just want to look at it by itself.

File: src/thindev.rs [hunk 1]
```diff
@@ -432,6 +440,38 @@ mod tests {
...
+
+    // Return a hashmap of key-value pairs for udev entry.
+    fn get_udev_db_entry(dev_node_search: &PathBuf) -> Option<HashMap<String, String>> {
+        // Takes a libudev device entry and returns the properties as a HashMap.
+        fn device_as_map(device: &libudev::Device) -> HashMap<String, String> {
+            let rc: HashMap<_, _> = device
+                .properties()
+                .map(|i| {
+                    (
+                        String::from(i.name().to_str().expect("Unix is utf-8")),
```
Comment (mulkieran) [inline src/thindev.rs]: Please just ```unwrap()``` here. That these property values can be decoded is far from a logical certainty. It seems like half the bugs that pyudev gets are about the fact that it's not even true. I'm not absolutely sure that the names can be decoded, although I feel that's likely.

Comment (tasleson) [reply src/thindev.rs]: Done, thanks!

**Pull request merged**